keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -32,8 +32,8 @@ class DeliveryConfigController(
     consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
-  @PreAuthorize("""@authorizationSupport.hasApplicationPermission('WRITE', 'DELIVERY_CONFIG', #name)
-    and @authorizationSupport.hasServiceAccountAccess('DELIVERY_CONFIG', #name)"""
+  @PreAuthorize("""@authorizationSupport.hasApplicationPermission('WRITE', 'APPLICATION', #deliveryConfig.application)
+    and @authorizationSupport.hasServiceAccountAccess(#deliveryConfig.serviceAccount)"""
```

Comment:
This was the fix: we can't read information about the delivery config from the database because it doesn't exist yet, so we get it from the submitted object.